### PR TITLE
Do not unnecessarily fetch stories/taxonomies

### PR DESCRIPTION
**Pull request by swissspidy**
## Context

When testing #12626 but also otherwise, I noticed superfluous HTTP requests for fetching stories and taxonomies when not even visiting the dashboard itself.

## Summary

Do not unnecessarily fetch stories/taxonomies when directly visiting a sub page like the settings page.

## User-facing changes

N/A

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Visit `/wp-admin/edit.php?post_type=web-story&page=stories-dashboard#/editor-settings` directly
2. See no HTTP requests for fetching stories and taxonomies
3. Switch to Dashboard
4. See stories & taxonomies filters loading

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #

### CODE REVIEW

**Comment by swissspidy** [comment]
Any suggestions regarding the failing karma tests are welcome

**Comment by miina** [comment]
@swissspidy I wanted to test the PR to see if the Karma tests failures might be valid but can't open the Explore templates at all currently, crashes with the following error:
Screenshot 2022-11-10 at 14 29 13

Screenshot 2022-11-10 at 14 31 44


Is this something you've also seen, or works ok for you?

**Comment by swissspidy** [comment]
Haven‘t seeb it yet, sounds like a bug I introduced 🙃

File: packages/dashboard/src/app/views/myStories/index.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ import useApi from '../../api/useApi';
 import useStoryFilters from './filters/useStoryFilters';
 import Content from './content';
 import Header from './header';
+import StoryFiltersProvider from './filters/StoryFiltersProvider';
```
Comment (spacedmonkey) [inline packages/dashboard/src/app/views/myStories/index.js:32]: Wondering if we should do this.
```
import { useStoryFilters, StoryFiltersProvider } from './filters';
```

**Comment by spacedmonkey** [comment]
Screenshot 2022-11-14 at 12 07 58

Also seeing this error.

**Comment by timarney** [comment]
Seeing the "find" issue as well

```
const search = filters.find(({ key }) => key === 'search');
```

**Comment by spacedmonkey** [comment]
Is this ready for review again?

**Comment by swissspidy** [comment]
Yeah now it is.

**Comment by timarney** [comment]
Outside of I see e2e tests failing on 5.9 (Telemetry Banner › should not display the banner after it has been closed) --- testing locally looks good.

**Review by spacedmonkey — APPROVED** [review]

**Pull request merged**